### ISSUE TITLE
Bug in inline - code behaves differently after Spartanization

### ISSUE THREAD

**Issue by ortroyaner** [SpartanRefactoring/Spartanizer#1089]
After running this code (before Spartanization):
```
ToggleGroup group = new ToggleGroup();
(new SelectAnArea()).getAllPossibleColors().forEach(c -> {
		JFXRadioButton rbtn = new JFXRadioButton(c);
		rbtn.setToggleGroup(group);
		});
```
I have a group of radio buttons which enables to choose only one radio button at a time.

Spartanizer suggest this code instead:
```
(new SelectAnArea()).getAllPossibleColors().forEach(c -> {
			JFXRadioButton rbtn = new JFXRadioButton(c);
			rbtn.setToggleGroup((new ToggleGroup()));
		});
```
The fact that a new ToggleGroup() is created every iteration within the forEach, bind each radio button to a different group, and therefore the limitation of only one radio button selected is gone.

TL;DR:

- Should not change it to new ToggleGroup() because the group should be the same in each iteration.

**Comment by yossigil** [comment]
@ravivos : can you please deal or pass?

**Comment by yuvalsimon** [comment]
@ravivos It might have been solved with #1067

**Comment by tomerdragucki** [comment]
still don't work for:
```java
trimmingOf("Object o = new Object();" + "l.forEach(c -> a(o));").stays();
```
because of DeclarationInlineIntoNext. problem in DeclarationInitializerStatementTerminatingScope solved.
tried to use DeclarationInitializerStatementTerminatingScope.never in DeclarationInlineIntoNext and broke tests. might try again.

**Comment by yossigil** [comment]
Yes. We have an issue with inlining. Working on it.
